### PR TITLE
Feature/ignore unknown wdl methods

### PR DESCRIPTION
**Pull request by denis-yuen**
Possible solution for #2139 

* the specific problem is that wdl4s doesn't understand wdl some 1.0 methods created afterwards
* without major changes (i.e. a big switch and update to womtool?) a hacky temporary solution is to assume that people using custom methods know what they are doing and allow them to pass validation, this does break the dag tool display for those specific workflows though as a heads-up

### CODE REVIEW

File: dockstore-common/src/main/scala/io/dockstore/common/Bridge.scala [hunk 1]
```diff
@@ -189,15 +192,22 @@ class Bridge(basePath : String) {
...
+      allTasks.foreach(taskTuple => {
+        val dockerAttributes = taskTuple._2.runtimeAttributes.attrs.get("docker")
+        tasks.put("dockstore_" + taskTuple._1.unqualifiedName, if (dockerAttributes.isDefined) dockerAttributes.get.collectAsSeq(passthrough).map(x => x.toWdlString.replaceAll("\"", "")).mkString("") else null)
+      })
+      tasks
+    } catch {
+      case ex: NoSuchMethodException =>
+        //FIXME: the best we can do is be generous and assume that unknown methods are WDL 1.0 methods until we update
+        // https://github.com/ga4gh/dockstore/issues/2139
+      null
```
Comment (coverbeck) [inline dockstore-common/src/main/scala/io/dockstore/common/Bridge.scala]: Return an empty LinkedHashMap instead? It seems like you'll get an NPE later in `mapConverterToToolInfo`? Or does the code handle an NPE somewhere further up the chain and that is OK?

File: dockstore-common/src/main/scala/io/dockstore/common/Bridge.scala [hunk 2]
```diff
@@ -240,18 +250,25 @@ class Bridge(basePath : String) {
...
         }
+        dependencyMap.put("dockstore_" + call.unqualifiedName, dependencies)
       }
-      dependencyMap.put("dockstore_" + call.unqualifiedName, dependencies)
+      dependencyMap
+    } catch {
+      case ex: NoSuchMethodException =>
+        //FIXME: the best we can do is be generous and assume that unknown methods are WDL 1.0 methods until we update
+        // https://github.com/ga4gh/dockstore/issues/2139
+        null
```
Comment (coverbeck) [inline dockstore-common/src/main/scala/io/dockstore/common/Bridge.scala]: Same as previous comment.

**Comment by denis-yuen** [comment]
From @coverbeck 's PR feedback, I also noticed that processing of imports will be broken too. 
Wondering to what extent this will be confusing to explain to users if imports, DAG processing, and validation is broken with WDL 1.0 as opposed to disallowing WDL 1.0 (which would suck for other reasons).

**Comment by denis-yuen** [comment]
On the other other hand, maybe we can be forgiven https://github.com/broadinstitute/cromwell/issues/4234

**Comment by denis-yuen** [comment]
> Is there a warning message we can surface for the case that we allow validation for a custom method that would've otherwise failed? Would help in tracking which cases/wdls are effected

It looks like the UI validation code doesn't currently have the concept of "valid, but with warnings". But I'll create an issue for us to follow-up while setting up the back-end appropriately.

**Comment by denis-yuen** [comment]
Screenshot from 2019-03-11 15-31-42

good news, this seems to work

**Pull request merged**